openedx/adg/lms/applications/forms.py

```diff
@@ -211,8 +211,6 @@ def clean(self):
         business_line_prerequisite = self.cleaned_data['business_line_prerequisite']
         is_common_business_line_prerequisite = self.cleaned_data['is_common_business_line_prerequisite']
 
-        if is_common_business_line_prerequisite and is_program_prerequisite:
+        if (business_line_prerequisite and (is_program_prerequisite or is_common_business_line_prerequisite)
+                or (is_common_business_line_prerequisite and is_program_prerequisite)):
             self.add_error('is_program_prerequisite', COURSE_GROUP_PREREQ_VALIDATION_ERROR)
-
```

Comment:
If business_line_prerequisite and is_common_business_line_prerequisite are checked, then the error would be displayed on `is_program_prerequisite` field. 
Also, if you still want to go with this, you can also do
`if bool(business_line_prerequisite) + is_common_business_line_prerequisite + is_program_prerequisite > 1:`

Reply:
That is a great suggestion Javeria!

Reply:
And yes, after discussing with QA, we only want the error displayed at the topmost field i.e `is_program_prerequisite`